### PR TITLE
Adding some functionality, correcting some errors

### PR DESCRIPTION
**Pull request by rowrowrowrow**

### CODE REVIEW

**Comment by d34dman** [comment]
I wish i had paid attention to notifications in the weekend. I pushed a fix for group, after realising the bug in a personal project.

**Comment by rowrowrowrow** [comment]
No worries, I created the package called enhanced-drupal-jsonapi-params cause I wanted some extra functionality and needed it via npm sooner than later. 

https://github.com/rowrowrowrow/enhanced-drupal-jsonapi-params

Let me know if you see anything of use I'm happy to combine our work into your original repo... but mine might be too opinionated. I also have more changes I plan to make. 

Thanks btw!!!

File: src/conjunctions.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const Conjunctions = {
```
Comment (d34dman) [inline src/conjunctions.ts:1]: We could use this

File: src/index.ts [hunk 2]
```diff
@@ -99,15 +110,40 @@ export class DrupalJsonApiParams {
     return key;
   }
 
-  public getQueryString(): string {
+  public mergeQueryObject(params: Params): DrupalJsonApiParams {
```
Comment (d34dman) [inline src/index.ts:113]: I wanted to include such functionality. But after introducing support for `optimised` query, it became difficult to create various test cases. Hence, looked away.

**Pull request closed without merging**